Model/Source/InsurancePossibilities.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Magento\Model\Source;
+
+use Magento\Framework\Data\OptionSourceInterface;
+use MyParcelNL\Sdk\src\Model\Consignment\PostNLConsignment;
+
+class InsurancePossibilities implements OptionSourceInterface
+{
+    /**
+     * Options getter
+     *
+     * @return array
+     */
+    public function toOptionArray()
+    {
+        $array = [];
+
+        foreach ($this->toArray() as $key => $day) {
```

Comment:
Ik ben meer voorstander van `array_map` in dit soort gevallen, het is voor veel php'ers een beetje eng want het doet denken aan javascript, maar het is echt exact bedoeld voor dit:


```php
return array_map(function ($day, $key) {
    return [
        'value' => $key,
        'label' => $day,
    ];
}, $this->toArray(), array_keys($this->toArray()));
```
Let wel op de dit stukje, `array_keys($this->toArray())`, bij php's array_map wordt niet standaard de index/key als 2e param meegegeven, hierdoor wel...